plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -875,17 +891,47 @@ public void revokeRoles(Set<String> roles, Set<HivePrincipal> grantees, boolean
     }
 
     private void invalidatePartitionCache(String databaseName, String tableName)
+    {
+        invalidatePartitionCache(databaseName, tableName, Optional.empty());
+    }
+
+    private void invalidatePartitionCache(String databaseName, String tableName, Optional<Predicate<Optional<String>>> partitionPredicate)
```

Comment:
Unnecessary use of `Optional`, since `partition -> true` is equivalent.

Reply:
Wanted to avoid calling convertDomainToPartitionName once it's not required as predicate is always true.
But yes it seems to be overkill.